pkg/plugin/instance/promql/querier.go

```diff
@@ -71,13 +72,33 @@ func (querier *Querier) QueryData(ctx context.Context, req *backend.QueryDataReq
 		Responses: backend.Responses{},
 	}
 
+	var (
+		m  sync.Mutex
+		wg sync.WaitGroup
+
+		// The default number of parallel queries is 10.
+		querierLimit = make(chan struct{}, 10)
```

Comment:
Is it too small(`10`) for the concurrency limit? Or can we make it configurable?

Reply:
Same number of parallel queries as Grafana:
![image](https://github.com/user-attachments/assets/eb0b770c-e8cc-4f0f-88e1-7f4b49692c54)